pkg/trace/api/api.go

```diff
@@ -182,11 +182,14 @@ func (r *HTTPReceiver) httpHandle(fn http.HandlerFunc) http.HandlerFunc {
 
 func (r *HTTPReceiver) httpHandleWithVersion(v Version, f func(Version, http.ResponseWriter, *http.Request)) http.HandlerFunc {
 	return r.httpHandle(func(w http.ResponseWriter, req *http.Request) {
-		mediaType := getMediaType(req)
+		mediaType, err := getMediaType(req)
+		if err != nil {
+			log.Errorf("error parsing Content-Type header: %v", mediaType)
```

Comment:
If we actually behave properly without error (which seems to be the case? no mediaType means default path?), then this shouldn't be an error. There is an error only if that default path doesn't work. a `log.Debugf("missing Content-Type header, assuming xxxxx")` would be a better fit?

Reply:
In that case I'd say to remove the `error` return value completely, and in case there is an error, we `log` directly inside the `getMediaType` function and return the `application/json` default. WDYT?

Reply:
I made the change. PTAL. Don't be shy to ask for further changes 😄